### PR TITLE
refactor(Content): convert Content.js to TypeScript

### LINKED ISSUE #13582: Add TypeScript types to Content

### PR DESCRIPTION
**Pull request by lux01**
Closes #13582

Converted `UIShell/Content.js` to `UIShell/Content.ts` and added appropriate type definitions.

#### Changelog

**New**

- Added an exported type for Content props

**Changed**

- Renamed `UIShell/Content.js` to `UIShell/Content.ts`

#### Testing / Reviewing

You can check the build still works and there are no errors when viewing UIShell storybook stories.

### CODE REVIEW

**Comment by lux01** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/UIShell/Content.ts [hunk 1]
```diff
@@ -7,15 +7,20 @@
...
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { HTMLAttributes, ReactNode } from 'react';
 import { usePrefix } from '../../internal/usePrefix';
 
+export type ContentProps = HTMLAttributes<HTMLElement> & {
+  children: ReactNode;
+  tagName?: string;
+};
+
```
Comment (tay1orjones) [inline packages/react/src/components/UIShell/Content.ts]: I know `className` is part of `HTMLAttributes`, but it'd be nice to have it explicitly specified here like `children`. Also all these are optional IIRC.

```suggestion
export type ContentProps = HTMLAttributes<HTMLElement> & {
  children?: ReactNode;
  className?: string 
  tagName?: string;
};

```

**Review by tay1orjones — COMMENTED** [review]
Thanks for this! Just one minor suggestion

**Review by andreancardona — APPROVED** [review]
other than Taylor's comment, looks good to me!

**Comment by andreancardona** [comment]
@lux01 let us know if you have any questions regarding our comment, thanks!

File: packages/react/src/components/UIShell/Content.ts [hunk 2]
```diff
@@ -7,15 +7,21 @@
...
-import React from 'react';
+import React, { HTMLAttributes, ReactNode } from 'react';
 import { usePrefix } from '../../internal/usePrefix';
 
+export type ContentProps = HTMLAttributes<HTMLElement> & {
+  children?: ReactNode;
+  className?: string 
+  tagName?: string;
+};
+
```
Comment (tay1orjones) [inline packages/react/src/components/UIShell/Content.ts]:
```suggestion
export type ContentProps = HTMLAttributes<HTMLElement> & {
  /**
   * Provide children nodes to be rendered in the content container
   */
  children?: ReactNode;
  
  /**
   * Optionally provide a custom class name that is applied to the container
   */
  className?: string;
  
  /**
   * Optionally specify the tag of the content node. Defaults to `main`
   */
  tagName?: string;
};

```

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**